core/toolbox.js

```diff
@@ -439,6 +483,32 @@ Blockly.Toolbox.prototype.scrollToCategoryByName = function(name) {
   }
 };
 
+/**
+ * Scroll to a category by id.
+ * @param {string} id The id of the category to scroll to.
+ * @package
+ */
+Blockly.Toolbox.prototype.scrollToCategoryById = function(id) {
+  var scrollPositions = this.flyout_.categoryScrollPositions;
+  for (var i = 0; i < scrollPositions.length; i++) {
+    if (id === scrollPositions[i].categoryId) {
+      this.flyout_.setVisible(true);
+      this.flyout_.scrollTo(scrollPositions[i].position);
+      return;
+    }
+  }
+};
+
+/**
+ * Get a category by its index.
+ * @param  {number} index The index of the category.
+ * @return {Blockly.Toolbox.Category} the category.
+ */
+Blockly.Toolbox.prototype.getCategoryByIndex = function(index) {
+  if (!this.categoryMenu_.categories_) return;
```

Comment:
If you're trying to return null, do that explicitly and also describe it in the return description.